nncf/accuracy_aware_training/algo.py

```diff
@@ -0,0 +1,316 @@
+
+    def configure_optimizers(self):
+        self.optimizer, self.lr_scheduler = self.train_epoch_args.configure_optimizers_fn()
+
+    def reset_training(self):
+        self.configure_optimizers()
+        for param_group in self.optimizer.param_groups:
+            param_group['lr'] *= self._base_lr_reduction_factor_during_search
+        self.training_epoch_count = 0
+        self.best_val_metric_value = 0
+
+    def dump_checkpoint(self, model, compression_controller):
+        checkpoint_path = osp.join(self.checkpoint_save_dir, 'acc_aware_checkpoint_last.pth')
+        checkpoint = {
+            'epoch': self.cumulative_epoch_count + 1,
+            'state_dict': model.state_dict(),
+            'best_metric_val': self.best_val_metric_value,
+            'current_val_metric_value': self.current_val_metric_value,
+            'optimizer': self.optimizer.state_dict(),
+            'scheduler': compression_controller.scheduler.get_state()
+        }
+        torch.save(checkpoint, checkpoint_path)
+        if self.best_val_metric_value == self.current_val_metric_value:
+            best_path = osp.join(self.checkpoint_save_dir,
+                                 'acc_aware_checkpoint_best_'
+                                 'compression_rate_{comp_rate}.pth'.format(comp_rate=self.compression_rate_target))
+            copyfile(checkpoint_path, best_path)
+
+    def update_training_history(self, compression_rate, best_metric_value):
+        best_accuracy_budget = best_metric_value - self.minimal_tolerable_accuracy
+        self._compressed_training_history.append((compression_rate, best_accuracy_budget))
+
+    def load_best_checkpoint(self, model):
+        # load checkpoint with highest compression rate and positive acc budget
+        possible_checkpoint_rates = [comp_rate for (comp_rate, acc_budget) in self._compressed_training_history
+                                     if acc_budget >= 0]
+        best_checkpoint_compression_rate = max(possible_checkpoint_rates)
+        resuming_checkpoint_path = osp.join(self.checkpoint_save_dir,
+                                            'acc_aware_checkpoint_best_compression_rate_'
+                                            '{comp_rate}.pth'.format(comp_rate=best_checkpoint_compression_rate))
+        nncf_logger.info('Loading the best checkpoint found during training '
+                         '{}...'.format(resuming_checkpoint_path))
+        resuming_checkpoint = torch.load(resuming_checkpoint_path, map_location='cpu',
+                                         pickle_module=restricted_pickle_module)
+        resuming_model_state_dict = resuming_checkpoint.get('state_dict', resuming_checkpoint,
+                                                            pickle_module=restricted_pickle_module)
+        load_state(model, resuming_model_state_dict, is_resume=True)
+
+
+# pylint: disable=E1101
+def run_accuracy_aware_compressed_training(model, compression_controller, nncf_config,
```

Comment:
It looks like the run_accuracy_aware_compressed_training function and below should be moved to the common part.

Reply:
agreed, done